plugins/inputs/infiniband/infiniband_nonlinux.go

```diff
@@ -0,0 +1,21 @@
+// +build !linux
+
+package infiniband
+
+import (
+	"log"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+func (e *Infiniband) Gather(acc telegraf.Accumulator) error {
+	return nil
+}
+
+func init() {
+	inputs.Add("infiniband", func() telegraf.Input {
+		log.Print("W! [inputs.infiniband] Current platform is not supported")
```

Comment:
Move this into a `Init()` function, as we noticed that this placement causes the warning to be printed at every startup.

https://github.com/influxdata/telegraf/blob/master/plugins/inputs/ethtool/ethtool_notlinux.go

Also, a bit of a nitpick but can you call this file: `infiniband_notlinux.go` for improved consistency.

Reply:
Sure, done! Don't worry about nitpicking, I'm always happy to learn a better way to do things :)